util/cloud/s3.h

```diff
@@ -18,7 +18,7 @@ using ListBucketsResult = io::Result<std::vector<std::string>>;
 // Empty if no more pages left.
 using ListObjectsResult = io::Result<std::string>;
 
-ListBucketsResult ListS3Buckets(const AWS& aws, http::Client* http_client);
+ListBucketsResult ListS3Buckets(AWS& aws, http::Client* http_client);
```

Comment:
it could be this but I prefer to pass AWS by pointer (style guide thingy).

Reply:
> Also comment that we can be updated with a refreshed token in some cases during the operation run
Why here? Isn't this a property of all AWS operations now?

Reply:
i mean, if you pass AWS by pointer, just explain what's the side effect of it. It's not clear from the signature why AWS is mutable,